igor-web/src/main/groovy/com/netflix/spinnaker/igor/gitlabci/service/GitlabCiService.java

```diff
@@ -69,13 +77,23 @@ public BuildServiceProvider getBuildServiceProvider() {
   }
 
   @Override
-  public GenericBuild getGenericBuild(String job, int buildNumber) {
-    throw new UnsupportedOperationException();
+  public GenericBuild getGenericBuild(String projectId, int pipelineId) {
+    Project project = client.getProject(projectId);
+    if (project == null) {
+      log.error("Could not find Gitlab CI Project with projectId={}", projectId);
+      return null;
+    }
+    Pipeline pipeline = client.getPipeline(projectId, pipelineId);
+    if (pipeline == null) {
+      return null;
+    }
+    return GitlabCiPipelineUtils.genericBuild(
+        pipeline, this.hostConfig.getAddress(), project.getPathWithNamespace());
   }
 
   @Override
-  public List<GenericBuild> getBuilds(String job) {
-    throw new UnsupportedOperationException();
+  public List<Pipeline> getBuilds(String job) {
+    return this.client.getPipelineSummaries(job, 25);
```

Comment:
Sorry I didn't see these earlier.  How'd you choose 25?  Yet another thing that would benefit from a config knob?

Reply:
I think it was just arbitrary to be honest.  Updated to use config option (default 100 for max page size) same as other pagination params.